app/controllers/api/v1/base_controller.rb

```diff
@@ -0,0 +1,60 @@
+class Api::V1::BaseController < ApplicationController
+  protect_from_forgery with: :null_session
+
+  before_action :skip_session
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found_handler
+  rescue_from ActiveRecord::RecordNotSaved, with: :not_acceptable_handler
+  rescue_from ActiveRecord::RecordInvalid, with: :not_acceptable_handler
+  rescue_from Api::Errors::PermissionRequired, with: :permission_required_handler
+
+  private
+
+  def authenticate
+    doorkeeper_authorize!
+    super if current_user
+  end
+
+  # @return [User] Current user, or +nil+ if no valid token.
+  def current_user
+    @current_user ||= User.undeleted.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
+  end
+
+  # @return [Ordergroup] Current user's ordergroup, or +nil+ if no valid token or user has no ordergroup.
+  def current_ordergroup
+    current_user.try(:ordergroup)
+  end
+
+  def require_ordergroup
+    authenticate
+    current_user.ordergroup or raise Api::Errors::PermissionRequired
```

Comment:
what about using `current_ordergroup`? do we use the `something or anotherthing` pattern anywhere else? why not `||`?

Reply:
Regarding `or`, I'd prefer that for flow-statements - [SO](https://stackoverflow.com/questions/2083112).

Reply:
We don't really use this pattern, I'll express it differently.

Reply:
Ah, now I understand why I did this: consider the following two pieces of code:
```ruby
raise Api::Errors::PermissionRequired unless current_ordergroup.present?
```
vs.
```ruby
current_ordergroup.present? or raise Api::Errors::PermissionRequired
```

In the former I first read an exception, and then see it only happens in some deviating circumstance. In the latter I first read that an ordergroup is present, and if not, the error is raised.
I think the latter is easier to read, because the main code-path is read first.

An alternative and (in my eyes) clear, but longer, way would be:
```ruby
unless current_ordergroup.present?
   raise Api::Errors::PermissionRequired
end
```

What do you think?

Reply:
i personally would prefer `raise Api::Errors::PermissionRequired unless current_ordergroup.present?` or the longer version.
imho this would also make clear that the return value is garbage, which was not clear for me in the first version

Reply:
Ah, that's indeed a pitfall. I'll adopt the version starting with `raise`.